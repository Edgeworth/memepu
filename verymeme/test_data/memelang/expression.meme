// status: ok
// input: ''
// output: ''
// returns: 42
fn main() i32 {
  -(34 + 32) * (55 / 3) - -4;
  a- - 4;
  a--- b;
  ---a+++b---c;
  ret 42;
}
