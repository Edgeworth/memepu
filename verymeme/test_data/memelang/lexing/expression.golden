Token(FN, 1:0, 'fn', 'fn')
Token(IDENT, 1:3, 'main', 'main')
Token(LPAREN, 1:7, '(', '(')
Token(RPAREN, 1:8, ')', ')')
Token(IDENT, 1:10, 'i32', 'i32')
Token(LBRACE, 1:14, '{', '{')
Token(MINUS, 2:5, '-', '-')
Token(LPAREN, 2:6, '(', '(')
Token(INT_LIT, 2:7, '34', 34, '34')
Token(PLUS, 2:10, '+', '+')
Token(INT_LIT, 2:12, '32', 32, '32')
Token(RPAREN, 2:14, ')', ')')
Token(ASTERISK, 2:16, '*', '*')
Token(LPAREN, 2:18, '(', '(')
Token(INT_LIT, 2:19, '55', 55, '55')
Token(FSLASH, 2:22, '/', '/')
Token(INT_LIT, 2:24, '3', 3, '3')
Token(RPAREN, 2:25, ')', ')')
Token(MINUS, 2:27, '-', '-')
Token(MINUS, 2:29, '-', '-')
Token(INT_LIT, 2:30, '4', 4, '4')
Token(SEMICOLON, 2:31, ';', ';')
Token(IDENT, 3:5, 'a', 'a')
Token(DMINUS, 3:6, '--', '--')
Token(INT_LIT, 3:9, '4', 4, '4')
Token(SEMICOLON, 3:10, ';', ';')
Token(IDENT, 4:5, 'a', 'a')
Token(DMINUS, 4:6, '--', '--')
Token(MINUS, 4:8, '-', '-')
Token(IDENT, 4:10, 'b', 'b')
Token(SEMICOLON, 4:11, ';', ';')
Token(DMINUS, 5:5, '--', '--')
Token(MINUS, 5:7, '-', '-')
Token(IDENT, 5:8, 'a', 'a')
Token(DPLUS, 5:9, '++', '++')
Token(PLUS, 5:11, '+', '+')
Token(IDENT, 5:12, 'b', 'b')
Token(DMINUS, 5:13, '--', '--')
Token(MINUS, 5:15, '-', '-')
Token(IDENT, 5:16, 'c', 'c')
Token(SEMICOLON, 5:17, ';', ';')
Token(RET, 6:5, 'ret', 'ret')
Token(INT_LIT, 6:9, '42', 42, '42')
Token(SEMICOLON, 6:11, ';', ';')
Token(RBRACE, 7:1, '}', '}')
